lib/server.js

```diff
@@ -413,6 +437,43 @@ WalletService.prototype.joinWallet = function(opts, cb) {
 };
 
 /**
+ * Save copayer preferences for the current wallet/copayer pair.
+ * @param {Object} opts
+ * @param {string} opts.email - Email address for notifications.
+ */
+WalletService.prototype.savePreferences = function(opts, cb) {
+  var self = this;
+
+  opts = opts || {};
+
+  self._runLocked(cb, function(cb) {
+    var preferences = Model.Preferences.create({
+      walletId: self.walletId,
+      copayerId: self.copayerId,
+      email: opts.email,
```

Comment:
lets use a popular module like https://github.com/Sembiance/email-validator


On Mon, May 11, 2015 at 11:39 AM, Ivan Socolsky <notifications@github.com>
wrote:

> In lib/server.js
> <https://github.com/bitpay/bitcore-wallet-service/pull/207#discussion_r30043833>
> :
>
> > @@ -413,6 +437,43 @@ WalletService.prototype.joinWallet = function(opts, cb) {
> >  };
> >
> >  /**
> > + * Save copayer preferences for the current wallet/copayer pair.
> > + * @param {Object} opts
> > + * @param {string} opts.email - Email address for notifications.
> > + */
> > +WalletService.prototype.savePreferences = function(opts, cb) {
> > +  var self = this;
> > +
> > +  opts = opts || {};
> > +
> > +  self._runLocked(cb, function(cb) {
> > +    var preferences = Model.Preferences.create({
> > +      walletId: self.walletId,
> > +      copayerId: self.copayerId,
> > +      email: opts.email,
>
> I will be adding a validation for the max length. Validating the actual
> email address is a hard problem to solve (almost all regexes tag valid
> addresses as invalid, unicode email addresses, etc). Let's just trust the
> user to provide a valid email address.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitpay/bitcore-wallet-service/pull/207/files#r30043833>
> .
>



-- 
Matías Alejo Garcia
@ematiu
Roads? Where we're going, we don't need roads!